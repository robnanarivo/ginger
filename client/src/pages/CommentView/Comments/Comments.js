import React from 'react';
import Stack from '@mui/material/Stack';
import Comment from './Comment';

function generateComments(comments, updatePost, admins) {
  const res = [];
  comments.slice().reverse().forEach((comment) => {
    res.push(<Comment {...comment} updatePost={updatePost} admins={admins} key={comment.id} />);
  });
  return res;
}

const Comments = (props) => {
  const { comments, updatePost, admins } = props;
  // const demo = [
  //   {
  //     body: 'wow the sky so beautiful!!!', avatar: 'L', date: 'September 16, 2021', name: 'Leo',
  //   },
  //   {
  //     body: 'Love your pic~', avatar: 'J', date: 'September 18, 2021', name: 'Jack',
  //   },
  //   {
  //     body: 'Thanks for sharing!', avatar: 'M', date: 'September 21, 2021', name: 'Tommy',
  //   },
  // ];
  return (
    <Stack spacing={2} sx={{ minWidth: '600px' }}>
      {generateComments(comments, updatePost, admins)}
    </Stack>
  );
};

export default Comments;
