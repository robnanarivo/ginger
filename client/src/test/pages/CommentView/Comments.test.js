import React from 'react';
import renderer from 'react-test-renderer';
import Comments from '../../../pages/CommentView/Comments/Comments';
import 'jest-styled-components';

describe('Comments unit test', () => {
  it('Comments Snapshot test', () => {
    const admins = [
      { id: 123, userName: 'goodboy' },
    ];

    const comments = [
      {
        id: 333, body: 'wow the sky so beautiful!!!', avatar: 'L', date: 'September 16, 2021', name: 'Leo',
      },
    ];

    const tree = renderer
      .create(<Comments admins={admins} comments={comments} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
