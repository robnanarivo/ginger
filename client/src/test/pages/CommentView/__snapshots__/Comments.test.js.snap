// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments unit test Comments Snapshot test 1`] = `
<div
  className="css-n481zt-MuiStack-root"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1nieira-MuiPaper-root-MuiCard-root"
  >
    <div
      className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-x7yxgc-MuiAvatar-root"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </div>
      <div
        className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          className="MuiTypography-root MuiTypography-body2 MuiCardHeader-subheader css-83ijpv-MuiTypography-root"
        >
          Invalid Date
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      />
    </div>
    <div
      className="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
    >
      
    </div>
  </div>
</div>
`;
